here-naksha-lib-core/src/main/java/com/here/naksha/lib/core/NakshaVersion.java

```diff
@@ -82,15 +106,31 @@ public NakshaVersion(int major, int minor, int revision) {
   public static @NotNull NakshaVersion of(@NotNull String version) throws NumberFormatException {
     final int majorEnd = version.indexOf('.');
     final int minorEnd = version.indexOf('.', majorEnd + 1);
-    return new NakshaVersion(
-        Integer.parseInt(version.substring(0, majorEnd)),
-        Integer.parseInt(version.substring(majorEnd + 1, minorEnd)),
-        Integer.parseInt(version.substring(minorEnd + 1)));
+    final int revisionEnd = version.indexOf('-', minorEnd + 1);
+    final int preReleaseTagEnd = revisionEnd == -1 ? -1 : version.indexOf('.', revisionEnd + 1);
+    final int major = Integer.parseInt(version.substring(0, majorEnd));
+    final int minor = Integer.parseInt(version.substring(majorEnd + 1, minorEnd));
+    final int revision = revisionEnd == -1
+        ? Integer.parseInt(version.substring(minorEnd + 1))
+        : Integer.parseInt(version.substring(minorEnd + 1, revisionEnd));
+    final String preReleaseTagString =
+        revisionEnd == -1 ? null : version.substring(revisionEnd + 1, preReleaseTagEnd);
+    final PreReleaseTag preReleaseTag =
+        preReleaseTagString == null ? null : PreReleaseTag.valueOf(preReleaseTagString);
+    final Byte preReleaseVersion =
+        preReleaseTagEnd == -1 ? null : Byte.parseByte(version.substring(preReleaseTagEnd + 1));
+
+    return new NakshaVersion(major, minor, revision, preReleaseTag, preReleaseVersion);
   }
 
   @AvailableSince(v2_0_3)
   public NakshaVersion(long value) {
-    this((int) ((value >>> 32) & 0xffff), (int) ((value >>> 16) & 0xffff), (int) (value & 0xffff));
+    this(
+        (int) ((value >>> 32) & 0xffff),
+        (int) ((value >>> 16) & 0xffff),
+        (int) (value & 0xffff),
+        (byte) ((value >>> 48) & 0xff) == 0 ? null : PreReleaseTag.valueOf((byte) ((value >>> 48) & 0xff)),
+        (byte) ((value >>> 48) & 0xff) == 0 ? null : (byte) ((value >>> 56) & 0xff));
```

Comment:
Can we change this slightly? Making the pre-release version an integer (this simplifies handling), so that it represents a value between 0 and 255 and not between -128 and 127? Then, store the pre-release tag and version in the lowest 16-bit instead of the highest 16-bit to keep the binary comparable? Last, set the pre-release tag and version by default to 255 (0xff), so that this represents a "final" version. Actually this means that 3.0.0.alpha-0 will be less than 3.0.0, which I think is right. About this last point we could discuss, maybe we say, that only 3.0.1 should be bigger than 3.0.0.alpha-0, while 3.0.0.alpha-0 should be bigger than 3.0.0, but I personally think it should be differnt, but just IMHO.

Reply:
Done.